src/vs/workbench/parts/explorers/browser/treeExplorerActions.contribution.ts

```diff
@@ -0,0 +1,60 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import * as nls from 'vs/nls';
+import { TPromise } from 'vs/base/common/winjs.base';
+import { Registry } from 'vs/platform/platform';
+import { IWorkbenchActionRegistry, Extensions as ActionExtensions } from 'vs/workbench/common/actionRegistry';
+import { SyncActionDescriptor } from 'vs/platform/actions/common/actions';
+import { Action } from 'vs/base/common/actions';
+import { IQuickOpenService, IPickOpenEntry } from 'vs/workbench/services/quickopen/common/quickOpenService';
+import { IActivityService } from 'vs/workbench/services/activity/common/activityService';
+
+const registry = Registry.as<IWorkbenchActionRegistry>(ActionExtensions.WorkbenchActions);
+
+export class ToggleExternalViewletAction extends Action {
+	public static ID = 'workbench.action.customTreeExplorer.toggle';
+	public static LABEL = nls.localize('toggleCustomExplorer', 'Toggle Custom Explorer');
+
+	constructor(
+		id: string,
+		label: string,
+		@IQuickOpenService private quickOpenService: IQuickOpenService,
+		@IActivityService private activityService: IActivityService
+	) {
+		super(id, name);
+	}
+
+	run(): TPromise<any> {
+		const viewletsToggleStataus = this.activityService.getIsEnabledForRegisteredViewlets();
+
+		const picks: IPickOpenEntry[] = [];
+		for (let viewletId in viewletsToggleStataus) {
+			picks.push({
+				id: viewletId,
+				label: (viewletsToggleStataus[viewletId] ? 'Disable ' : 'Enable ') + this.getShortViewletId(viewletId)
```

Comment:
Is this `viewletId` the `treeExplorerNodeProviderId` in the contribution? If so why aren't you using treeLabel as the label in the quick open?